tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+  ObAdminBackupPieceLogIterationTask *sibling_task = nullptr;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_ISNULL(piece_validation_dag
+                       = static_cast<ObAdminBackupPieceValidationDag *>(get_dag()))) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "failed to get piece validation dag", K(ret));
+  } else if (0 == task_id_) {
+    ObAdminPieceKey backup_piece_key;
+    ObAdminBackupPieceValidationAttr *backup_piece_attr = nullptr;
+    if (OB_FAIL(ctx_->processing_backup_piece_key_array_.at(piece_validation_dag->get_id(),
+                                                            backup_piece_key))) {
+      STORAGE_LOG(WARN, "failed to get backup piece key", K(ret),
+                  K(piece_validation_dag->get_id()));
+    } else if (OB_FAIL(ctx_->get_backup_piece_attr(backup_piece_key, backup_piece_attr))) {
+      STORAGE_LOG(WARN, "failed to get backup piece attr", K(ret), K(backup_piece_key));
+    } else if (OB_FAIL(backup_piece_attr->split_lsn_range(
+                   piece_validation_dag->processing_lsn_range_array_))) {
+      STORAGE_LOG(WARN, "failed to split lsn range", K(ret), K(backup_piece_key));
+    } else if (OB_FAIL(backup_piece_attr->stat_.add_scheduled_lsn_range_count_(
+                   piece_validation_dag->processing_lsn_range_array_.count()))
+               || OB_FAIL(ctx_->global_stat_.add_scheduled_lsn_range_count_(
+                   piece_validation_dag->processing_lsn_range_array_.count()))) {
+      STORAGE_LOG(WARN, "failed to add scheduled lsn range count", K(ret), K(backup_piece_key));
+    } else {
+      STORAGE_LOG(INFO, "succeed to split lsn range", K(ret), K(backup_piece_key));
+    }
+  }
+  if (OB_SUCC(ret)) {
+    if (next_task_id >= piece_validation_dag->processing_lsn_range_array_.count()) {
+      ret = OB_ITER_END;
+      next_task = nullptr;
+      STORAGE_LOG(INFO, "no more sibling task", K(ret), K(next_task_id));
+    } else if (OB_FAIL(piece_validation_dag->alloc_task(sibling_task))) {
+      STORAGE_LOG(WARN, "failed to alloc sibling task", K(ret));
+    } else if (OB_FAIL(sibling_task->init(next_task_id, ctx_))) {
+      STORAGE_LOG(WARN, "failed to init sibling task", K(ret));
+    } else {
+      next_task = sibling_task;
+      sibling_task = nullptr;
+    }
+  }
+
+  return ret;
+}
+int ObAdminBackupPieceLogIterationTask::process()
+{
+  int ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_ISNULL(ctx_) || ctx_->aborted_) {
+    ret = OB_CANCELED;
+    STORAGE_LOG(WARN, "aborted", K(ret));
+  } else if (OB_FAIL(iterate_log_())) {
+    STORAGE_LOG(WARN, "failed to iterate log", K(ret));
+  } else {
+    STORAGE_LOG(DEBUG, "succeed to process ObAdminBackupPieceLogIterationTask", K(ret),
+                K(task_id_));
+  }
+  return ret;
+}
+int ObAdminBackupPieceLogIterationTask::iterate_log_()
```

Comment:
If continuse between log files is checked?

Reply:
a piece is divided into multiple part without knowledge of specific log file, if log file is discontinue, we will encounter a failure